### ISSUE TITLE
separating commands by tags

### ISSUE THREAD

**Issue by jmls** [danielgtaylor/restish#224]
if I have an openapi spec that has a number of tags (let say "foo" and "bar") then when I do a `restish myapi -h` I get 

```
Usage:
  restish myapi [command]
  
  foo commands:
    update
    
  bar commands
    update
    
    
```

yes, my operation ids are the same (using nestjs and controllers, so generated from the method name)

I was hoping to be able to do something like

restish myapi foo update
restish myapi bar update

but it seems not.

Am I forced into changing the operationIds to be fooUpdate and barUpdate ? my restish api  then looks like this

```
Usage:
  restish myapi [command]
  
  foo commands:
    fooupdate
    
  bar commands
    barupdate
    
    
```
which kinda looks .. sucky ;) 

So in a nutshell, I 'm looking for some way of using the tags in the spec to specify a command and sub-command (tag / operratiionid)

**Comment by danielgtaylor** [comment]
@jmls great question. Right now Restish is set up to have a flat list of operation commands rather than a hierarchy like you are describing. There are a few different ways you could make a hierarchy and it's not easy to do this in a generic way that would work for everyone, but I'm open to ideas!

As for using the same operation ID multiple times... see the [OpenAPI spec](https://spec.openapis.org/oas/v3.1.0#operation-object):

> Unique string used to identify the operation. The id MUST be unique among all operations described in the API. The operationId value is case-sensitive. Tools and libraries MAY use the operationId to uniquely identify an operation, therefore, it is RECOMMENDED to follow common programming naming conventions.

I'm not sure what nestjs is doing or recommends, but the spec says they must be unique, which is why Restish uses them as the unique command name.

**Comment by danielgtaylor** [comment]
Thanks for the clear request. I checked this against v2.0.0, and tag-based command layout is available now.

In v2 you can set an API to tag layout:

```bash
restish api set myapi 'command_layout: tags'
```

I verified the exact shape from this issue with a small OpenAPI 3.1 spec containing `foo` and `bar` tags and duplicate `operationId: update` values. With tag layout enabled, v2 shows and registers separate nested commands:

```text
Examples:
  restish tagtest foo update
  restish tagtest bar update

Available Commands:
  bar         Operations tagged bar
  foo         Operations tagged foo
```

and each tag subcommand exposes its own `update` operation. I also verified the behavior with the focused v2 test:

```bash
go test ./internal/cli ./internal/spec -run 'TestGeneratedCommandLayoutTagsCreatesTagSubcommands'
```

I'm closing this as implemented in v2. Please give Restish v2.0.0 a try.
